### ISSUE TITLE
Can't get https working

### ISSUE THREAD

**Issue by matthewbrutonall** [cantaloupe-project/cantaloupe#431]
Hi,
I keep getting timeouts when trying to get HTTPS working. 
I created a keystore as per these instructions: https://ordina-jworks.github.io/security/2019/08/14/Using-Lets-Encrypt-Certificates-In-Java.html
And then referred to the keystore path in the properties of Cantaloupe.
Tried disabling http also. Didn't work. Tried changing the ports. Also didn't work. 
Everything works fine on http. 
I have no idea what else to try. Please help!

**Comment by dxu1789** [comment]
I am also having trouble getting HTTPS working. I followed instructions here, modified slightly for my situation: https://www.digicert.com/kb/csr-ssl-installation/tomcat-keytool.htm

I've pointed the Cantaloupe config to the keystore on the server, and when I start it up I can see it is now listening on port 8183 using **lsof**, in addition to the default 8182. However I cannot load the splash page like I can on the default port 8182. There is no mention of HTTPS in either the application log or error log. How can I debug this?

**Comment by adolski** [comment]
I don't have access to a domain with which to test a Let's Encrypt certificate, but I can generate a self-signed certificate and use it without any problems under 4.1.7:

```sh
$ keytool -genkey -keyalg RSA -alias mykey -keystore keystore.jks -storepass password -validity 360 -keysize 2048
```

**cantaloupe.properties**
```properties
https.enabled = true
https.host = localhost
https.port = 8183
https.http2.enabled = true
https.key_store_type = JKS
https.key_store_password = password
https.key_store_path = /home/me/keystore.jks
https.key_password = password
```

```sh
curl -k https://localhost:8183
```

If you aren't seeing anything in the log, and are able to build using Maven, see if removing this `if` clause helps: https://github.com/cantaloupe-project/cantaloupe/blob/db656d124cadedd1abf5e7b55fb5dbafa3f3f2fa/src/main/java/edu/illinois/library/cantaloupe/logging/ApplicationLogFilter.java#L24-L29

**Comment by dxu1789** [comment]
I was able to get this working and wrote up my process which I was able to repeat. 
https://gist.github.com/dxu1789/64306e34aa7bd14f68f6869320c25719

One tricky thing I ran into was that one of my servers had the IBM version of Java installed, while my keystore was generated on my workstation which was OpenJDK. When I tried to run keytool -list on my server it threw an error, which clued me in to the source of the problem. Removing that Java and installing OpenJDK fixed it.
